src/_imaging.c

```diff
@@ -3534,6 +3534,34 @@ _save_ppm(ImagingObject *self, PyObject *args) {
     return Py_None;
 }
 
+static PyObject *
+_set_internal_pixel_bytes(ImagingObject *self, PyObject *args) {
```

Comment:
Personally I don't think this is a good idea to add core method just for testing. The problems are:
* It needs maintenance in future
* It exposes internals which will be harder to change
* This method itself doesn't tested

However, for a long time I suffer from lack of functionality just to change internal model of the image without copying. Probably we can add something like `ImagingCore.rewrite_mode()` which will work within the same `pixelsize`.

Reply:
> It exposes internals which will be harder to change

I don't think it exposes anything that wouldn't also be exposed by `ImagingCore.rewrite_mode()`.

> This method itself doesn't tested

That can be added.

Reply:
> Probably we can add something like `ImagingCore.rewrite_mode()` which will work within the same `pixelsize`.

Unfortunately, pixelsize/linesize are not available anywhere except on an image - there is no list to loop through that has this information.

Reply:
Damn, right. Maybe create minimal image with target mode just to check it's pixel size?

Reply:
I think that would work. I still don't see how this is any better than the function I already added though.

Reply:
It have practical use cases outside of tests at least

Reply:
I ran into another issue. `Image.mode` and `Image.im.mode` are not the same object. So I can't set `Image.mode` from a method on `ImagingCore`. There's a note to fix this, but it hasn't been done yet: https://github.com/python-pillow/Pillow/blob/f089c2db8c1806f09e6ed53a1986146693db1ade/src/PIL/Image.py#L483C40-L483C40

Reply:
I can make `Image.mode` a property, but `Image.mode` is currently writable, while `Image.im.mode` is not.

Reply:
I've created #7271 to look into that.

Reply:
To unblock this PR I'd be happy with tests for modes we can test without core changes.

Reply:
With `ImagingCore.rewrite_mode()` there are none, because `ImagingCore.rewrite_mode()` cannot update the mode that is currently returned from `Image.mode`.

Reply:
I've created https://github.com/Yay295/Pillow/pull/8 as a suggestion.